lib/commands/create.js

```diff
@@ -57,30 +60,11 @@ function build () {
       alias: 'o',
       description: 'the file to which to write cluster data, for debugging purposes'
     },
-    vaultHost: {
-      description: 'the host of the vault server containing sensitive cluster information, auth data, and defaults',
-      default: process.env.VAULT_HOST
-    },
-    vaultToken: {
-      description: 'an auth token for the vault server',
-      default: process.env.VAULT_TOKEN
-    },
-    vaultRoleId: {
-      description: 'an AppRole id token for the vault server',
-      default: process.env.VAULT_ROLE_ID
-    },
-    vaultSecretId: {
-      description: 'an AppRole secret id token for the vault server',
-      default: process.env.VAULT_SECRET_ID
-    },
-    vaultSecretPrefix: {
-      description: 'the prefix for the KV secrets engine',
-      default: process.env.VAULT_SECRET_PREFIX || 'kv/'
-    },
     provider: {
       description: 'the cloud provider to use, defaults to KUBE_SERVICE environment variable',
       default: process.env.KUBE_SERVICE || 'GKE'
-    }
+    },
+    ...commonArgs.vault()
```

Comment:
there's a way you can do this with yargs, tl;dr your top level command defines your shared options and then your sub commands define only their extra requirements. just wanted to share another approach though, not worth refactoring everything over

Reply:
TIL!  Although, I wonder how many other framework/library features out there can be more succinctly replaced with object spread...